### PR TITLE
QPPCT-1008 JsonWrapper Metadata Refactor

### PR DESCRIPTION
**Pull request by duselmann**
### Information
- JIRA story QPPCT-1008.

### Changes proposed in this PR:
- Refactors Metadata out of the JsonWrapper named string map.
- Uses the JsonWrapper recursively to hold object rather than Object type.
-

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [ ] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

**Comment by duselmann** [comment]
I submitted to allow eyes on it throughout the day. There are some quick followups to come.
I checked that this branch was up to date with develop last night and will merge in new code later today. Also, all tests passed and I "increased" coverage but apparently not enough. Will increase coverage as well.

File: converter/pom.xml [hunk 1]
```diff
@@ -111,5 +111,36 @@
 			<artifactId>cloning</artifactId>
 			<version>1.9.10</version>
 		</dependency>
+
+        <dependency>
```
Comment (omnivion) [inline converter/pom.xml]: these deps should all be automatically imported from test-commons, right?

Comment (duselmann) [reply converter/pom.xml]: They do not get pulled in properly with Eclipse JUnit engine. Not Sure why. But yes, this is not a problem with command line maven.

**Comment by duselmann** [comment]
There is a potential design improvement in the Path Correlation to be done (by me). The original impl worked with maps and filtered on metadata_holder. Now that this feature is fully passing all tests, it would be nice to refactor that to walk the new JsonWrapper metadata structure.

**Pull request merged**